tests/infra/SearchServiceImpl.ts

```diff
@@ -157,10 +160,13 @@ describe('SearchService', () => {
       const customUrl = 'http://custom.endpoint.com/api';
       const config: SearchConfigWithDefaulting = {
         ...configWithRequiredApiKey,
-        endpoints: {
-          ...defaultEndpoints,
-          universalSearch: customUrl
-        }
+        endpoints:
+          new EndpointsProvider(
```

Comment:
rather than making the EndpointsProvider public, I think it's worth considering making it internal and instead exporting a `provideEndpoints` function which calls the EndpointsProvider internally. That would follow the existing naming convention in our repos  e.g. (`provideCore` and `provideHeadless`), and it would be simpler since they wouldn't need to construct a class. For example, the call would look like `provideEndpoints(Environment.PROD, CloudRegion.US`.

Reply:
Done